tdrs-backend/tdpservice/users/admin.py

```diff
@@ -25,30 +25,6 @@ def clean(self):
         groups = cleaned_data['groups']
         if len(groups) > 1:
             raise ValidationError("User should not have multiple groups")
-        group = groups.first()
-        location_type = cleaned_data['location_type']
-        role_location_type_map = {
-            'OFA Regional Staff': 'region',
-            'Data Analyst': 'stt',
-            'Developer': 'stt'
-        }
-
-        # allow saving the user without a group type
-        # updating a user's account approval status without assigning a group used to throw an error
-        # a group assignment can give the user access to some frontend features regardless of `account_approval_status`
-        # users in `Pending` status should not have a group assigned
-        if group is not None:
-            correct_location_type = role_location_type_map.get(group.name)
-            location_based_role = group.name in ('OFA Regional Staff', 'Data Analyst', 'Developer')
-
-            if (location_based_role and (location_type and location_type.name != correct_location_type)):
-
-                raise ValidationError("Incorrect location type for role")
```

Comment:
this logic has all been moved to the `User.validate_location` method. Actually, it was already there.